misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/ObjectMock.java

```diff
@@ -245,6 +365,158 @@ public static BError validateArguments(BObject caseObj) {
+                        }
+                        if (!isTypeAvailable) {
+                            return ErrorCreator.createError(
+                                    MockConstants.TEST_PACKAGE_ID,
+                                    MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                                    StringUtils.fromString(detail),
+                                    null,
+                                    new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                        }
+                    } else if (paramType instanceof ArrayType) {
+                        Object arg = it.next();
+                        if (!(TypeChecker.getType(arg) instanceof ArrayType)) {
+                            return ErrorCreator.createError(
+                                    MockConstants.TEST_PACKAGE_ID,
+                                    MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                                    StringUtils.fromString(detail),
+                                    null,
+                                    new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                        }
+                    } else if (!TypeChecker.checkIsType(it.next(), paramType)) {
+                        return ErrorCreator.createError(
+                                MockConstants.TEST_PACKAGE_ID,
+                                MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                                StringUtils.fromString(detail),
+                                null,
+                                new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                    }
+                }
+                break;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Validates the resource function arguments.
+     *
+     * @param caseObj ballerina object that contains information about the case to register
+     * @return an optional error if a validation fails
+     */
+    public static BError validateResourceArguments(BObject caseObj) {
+        GenericMockObjectValue genericMock = (GenericMockObjectValue) caseObj.getObjectValue(
+                StringUtils.fromString(MockConstants.MOCK_OBJECT));
+        String functionName = caseObj.getStringValue(StringUtils.fromString(MockConstants.FUNCTION_NAME)).toString();
+        BArray argsList = caseObj.getArrayValue(StringUtils.fromString(MockConstants.ARGS));
+        String functionPattern = getFunctionNameForResourcePath(functionName);
+        String accessor = caseObj.getStringValue(StringUtils.fromString(MockConstants.ACCESSOR)).toString();
+        functionPattern = MockConstants.DOLLAR_RESOURCE_SEPARATOR + accessor +
+                MockConstants.DOLLAR_RESOURCE_SEPARATOR + functionPattern;
+        for (ResourceMethodType attachedFunction : ((BClientType) genericMock.getType()).getResourceMethods()) {
+            if (attachedFunction.getName().endsWith(functionPattern)) {
```

Comment:
shall we return early `if !attachedFunction.getName().endsWith(functionPattern)`, then we can remove this if condition.

Reply:
We need to loop through all the `attachedFunctions` to know the match for the pattern. Early return may not be possible here